mempool/src/pool.rs

```diff
@@ -790,6 +801,25 @@ where
             .collect()
     }
 
+    fn collect_txs(
+        &self,
+        mut tx_accumulator: Box<dyn TransactionAccumulator>,
+    ) -> Vec<SignedTransaction> {
+        if self.store.is_empty() {
```

Comment:
I'm confused. Isn't it an invariant that anything that comes from tx_iter has a corresponding tx_id in the txs_by_id container?

In what situation could we have something in txs_by_descendant_score but not in txs_by_id?

Reply:
You're absolutely right.
What I overlooked is the fact that if the pool is empty the txs_by_descendant score iterator won't yield any elements. Removed the check since it's redundant.